### PR TITLE
Fix browser back from detail page

### PR DESCRIPTION
**Pull request by pratap0007**
- the problem was that we were updating redux store object 'TaskData' with selected task card's data itself so to fix it I have created a separate redux store object 'TaskDataList'.
- to remove console error add key in Filter component
- modification in SearchBar component to render all task card when users remove typed task name.

### CODE REVIEW

File: frontend/src/constants/index.ts [hunk 1]
```diff
@@ -10,8 +10,7 @@ window.config = window.config || {
 };
 
 export const ENTER_KEY = 13;
-export const API_URL = window.config.API_URL;
+export const API_URL =window.config.API_URL;
```
Comment (sthaha) [inline frontend/src/constants/index.ts]: unrelated change. could you please remove this?

**Review by sthaha — CHANGES_REQUESTED** [review]
Looks good to me but could you please remove unrelated changes?

File: frontend/src/constants/index.ts (hunk 1 above)
Comment (pratap0007) [reply frontend/src/constants/index.ts]: yeah will remove it.

File: frontend/src/components/basic-detail/BasicDetailParent.tsx [hunk 2]
```diff
@@ -3,31 +3,26 @@ import {useParams} from 'react-router';
...
   const {taskId} = useParams();
-
-
   React.useEffect(() => {
     props.fetchTaskName(taskId);
     // eslint-disable-next-line
   }, []);
-
-  if (props.TaskName != null) {
+  if (props.TaskName && (props.TaskName.id).toString() === taskId) {
```
Comment (sthaha) [inline frontend/src/components/basic-detail/BasicDetailParent.tsx:15]: does `?.` work here? `props.TaskName?.id.toString() === taskId` ?

Comment (pratap0007) [reply frontend/src/components/basic-detail/BasicDetailParent.tsx:15]: yeah it works.
